### ISSUE TITLE
Feature: Transpose optimization for Softmax

### ISSUE THREAD

**Issue by fthielke** [onnx/tensorflow-onnx#1716]
Thanks to the optimizations by tf2onnx, most Transpose operations in converted models are eliminated.
However, Softmax does not seem to be accounted for yet.

Take the following simple SavedModel:
```py
import tensorflow as tf
from keras import Model
from keras.layers import Input, Softmax
import keras.backend as K
K.set_image_data_format('channels_first')

inputs = Input((1, None, None), name='input')
outputs = Softmax(axis=1, name='output')(inputs)
m = Model(inputs, outputs)

tf.saved_model.save(
    obj=m,
    export_dir='test_softmax'
)
```

When exporting it via `python -m tf2onnx.convert --saved-model test_softmax --output test_softmax/model.onnx --tag serve --signature_def serving_default --opset 14`, the result looks like this:
![](https://user-images.githubusercontent.com/15705085/133253518-5ae3b5da-ff8c-4619-a4ae-69797f1b65a2.png)

Same for every other channels_first model I export - the Softmax operation is always surrounded by Transposes.

**Comment by TomWildenhain-Microsoft** [comment]
@fthielke  Yes, this would be a good optimization to have.  Do you think you could try to add it to the transpose optimizer and submit a PR? 
https://github.com/onnx/tensorflow-onnx/blob/446494eea2fb80b4beca13914811678399c91e11/tf2onnx/optimizer/transpose_optimizer.py#L224

**Comment by fatcat-z** [comment]
Wait for user to fix related PR issue.
